go/chat/crypto_doc.md

```diff
@@ -0,0 +1,180 @@
+message to Bob, it's possible for Bob to prove to other people that Alice sent
+it.
+
+**Reason:** This is a side effect of using signing keys to authenticate
+messages. Repudiable authentication works well in a chat between two people,
+and it's even cheaper than signing. But it doesn't scale well to group chats
+with tons of people, because the sender needs to authenticate each message
+separately for every other person in the group. It also makes it difficult to
+add new people to an existing group (which we might support in the future),
+because old messages can't be reauthenticated until all the senders come
+online.
+
+### History Integrity
+
+**Tradeoff:** It's possible for the server to change history in specific ways,
+like delaying messages or reordering senders who don't know about each other
+yet.
+
+**Reason:** Good performance in a messenger app requires letting different
+people send messages at the same time, without blocking either of them to
+decide who goes first, or forcing either of them to have the perfectly
+up-to-date message history. That means that the server will always have a lot
+of leeway in deciding what actually happened. The previous message references
+guarantee that the server can't put a message before one it was replying to, or
+delete messages after other people have seen them.
+
+### Metadata
+
+**Tradeoff:** Keybase is a centralized service, and the server receives a lot
+of metadata. It knows who's talking to whom and how much data they're sending
+back and forth. It also knows the type of each message, like "text",
+"attachment", or "deletion".
+
+**Reason:** Decentralized services make it difficult to add features over time,
+unless the developers can break backwards compatibility, which defeats most of
+the purpose of decentralizing. Moxie Marlinspike, the lead developer of Signal,
+has an [excellent
+article](https://whispersystems.org/blog/the-ecosystem-is-moving/) about these
+issues. Decentralized services can also be *easier* to spy on, depending on
+who's doing the spying. It's much easier to run a malicious Tor exit node than
+to break into Facebook's servers, for example.
+
+### Keys on Disk
+
+**Tradeoff:** Keybase sometimes keeps secret keys accessible on disk, instead
+of encrypting them with your password all the time.
+
+**Reason:** Modern operating systems make full-disk encryption convenient, so
+the use case for application-specific encrypted storage is much more limited
+than it used to be. It mostly assumes that an attacker can read arbitrary files
+from your disk but can't run arbitrary code. That can happen, but it's a very
+specific scenario, and it probably leaks your decrypted files and messages even
+if it doesn't leak your keys. By comparison, throwing extra password prompts at
+the user is a major downside for everyone all the time, especially non-experts
+who don't know how to pick good passwords or manage them.
```

Comment:
Perhaps add a summary sentence: "So we store unencrypted secrets derived from your passphrase on disk, make device key revocation easy, and encourage you to use full-disk encryption to protect these secrets at rest."